### PR TITLE
Generic ThingIFAPI and ThingIFAPIBuilder

### PR DESCRIPTION
**Pull request by yongpingchen**
The snippet to using generic for ThingIFAPI is like this:

```java

// TraitAlais API
        ThingIFAPI<TraitAlias> api = ThingIFAPIBuilder
                .<TraitAlias>newBuilder(
                        getApplicationContext(),
                        app,
                        owner
                .build();

        List<Action> actions = new ArrayList<>();
        TurnPower turnPower = new TurnPower(true);
        actions.add(turnPower);

        //Trait formatted command
        TraitAlias airConditionerAlias = new TraitAlias("AirConditionerAlias");
        List<Pair<TraitAlias, List<Action>>> traitActions = new ArrayList<>();
        traitActions.add(new Pair<>(airConditionerAlias, actions));
        CommandForm<TraitAlias> form = new CommandForm<>(traitActions);

        try {
            api.postNewCommand(form);
        }catch(Exception ex){
            //handle Exception
        }

// NonTraitAlias API
        ThingIFAPI<NonTraitAlias> api1 = ThingIFAPIBuilder
                .<NonTraitAlias>newBuilder(
                        getApplicationContext(),
                        app,
                        owner
                        )
                .build();

        //Non trait formatted command
        NonTraitAlias nonTraitAlias = new NonTraitAlias();
        List<Pair<NonTraitAlias, List<Action>>> nonTraitActions = new ArrayList<>();
        nonTraitActions.add(new Pair<>(nonTraitAlias, actions));
        CommandForm<NonTraitAlias> form1 = new CommandForm<>(nonTraitActions);

        try {
            api1.postNewCommand(form1);
        }catch(Exception ex){
            //handle Exception
        }
```

### CODE REVIEW

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 1]
```diff
@@ -51,7 +51,7 @@
 /**
  * This class operates an IoT device that is specified by {@link #onboard(String, String, String, JSONObject)} method.
  */
-public class ThingIFAPI implements Parcelable {
+public class ThingIFAPI<AliasType extends Alias> implements Parcelable {
```
Comment (shinsuke-nara) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: @yongpingchen 

This is cosmetic question. So this is not essential one.

In java, type parameter name is used to described by one uppercase letter like `T`.
Is `AliasType` good manner for Java?

Comment (yongpingchen) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: From the official [document](https://docs.oracle.com/javase/tutorial/java/generics/types.html) about java generics: 
> By convention, type parameter names are single, uppercase letters. 

Suggest to use a single uppercase letter.

**Comment by yongpingchen** [comment]
@shinsuke-nara  fixed by your comment. Please check

**Comment by shinsuke-nara** [comment]
@yongpingchen LGTM:+1:

**Pull request merged**